lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb

```diff
@@ -436,8 +436,12 @@ def cmd_channel(*args)
         return
       end
 
-      client.channels.each_pair do |id, channel|
-        cmd_close(id)
+      begin
+        client.channels.each_pair do |id, channel|
+          cmd_close(id)
+        end
+      rescue ::StandardError
+        print_line('Failed when trying to kill a channel.')
```

Comment:
which channel?

Reply:
Looks like this still has the same issue, i.e. if one individual channel kill fails - the remainder won't be closed